standards/bulk.md

```diff
@@ -0,0 +1,134 @@
+    "operations": [                                                         // there are some operations that failed and succeeded
+        {
+            "ordinal": 0,                                                   // position in original ordered operation collection
+            "action": "CREATE" | "UPDATE" | "CREATE_UPDATE" | "DELETE",     // repeat action type
+            "id": "string" | null,                                          // the associated id of the entity, if available
+            "ref": "sps-ref" | null,                                        // the associated sps-ref URN entity, if applicable
+            "status": "SUCCEEDED" | "FAILED",                               // status of individual operation
+            "detail": "string",                                             // indicates details on the operation result, such as a error message.
+        }
+    ]
+}
+```
+
+The following example demonstrates various operations in a bulk request and response:
+
+```
+// REQUEST
+PATCH /articles
+Content-Type: application/json
+{
+    "operations": [
+        {
+            "action": "CREATE_UPDATE"
+            "ifMatch": "33a64df551425fcc55e4d42a148795d9f25f89d4",
+            "entity": {
+                "id": "bfd8f0c0-be67-4f81-bf82-e55e552609f4",
+                "name": "my name",
+                "description": "my description"
+            }
+        },
+        {
+            "action": "CREATE"  
+            "entity": {
+                "id": null,
+                "name": "my name",
+                "description": "my description"
+            }
+        },
+        {
+            "action": "DELETE" 
+            "ifMatch": "44a64df551425fcc55e4d42a148795d9f25f89c5",     
+            "entity": {
+                "id": "d9bd5d91-fc25-4410-ae42-c8f631e8e9ff",
+                "name": null,
+                "description": null
+            }
+        },
+    ]
+}
+ 
+// RESPONSE
+200 OK
+Content-Type: application/json
+{
+    "status": "PARTIAL",                                 
+    "operations": [
+        {
+            "ordinal": 0,
+            "action": "CREATE_UPDATE",
+            "id": "bfd8f0c0-be67-4f81-bf82-e55e552609f4",
+            "ref": "sps:...",
+            "status": "SUCCEEDED",                         
+            "detail": "Article was updated.",  
+        },
+        {
+            "ordinal": 1,
+            "action": "CREATE",
+            "id": null,
+            "ref": null,
+            "status": "FAILED",                         
+            "detail": "Could not create article, since the name already exists.",  
+        },
+        {
+            "ordinal": 2,
+            "action": "DELETE",
+            "id": "d9bd5d91-fc25-4410-ae42-c8f631e8e9ff",
+            "ref": "sps:...",
+            "status": "SUCCEEDED",                         
+            "detail": null,  
```

Comment:
can the entire response payload be generic? or do we need to allow for additional extensions?

Reply:
I think it is likely that sometimes more detail would be useful.

Reply:
By more detail do you mean with regards to the "detail" field? or allow for extensions (other fields)?

Reply:
I think allowing a structure under detail should be sufficient.

Reply:
Moving to a proposed fully opinionated detail object. Coming soon.